### ISSUE TITLE
Improve Dropdown keyboard navigation

### ISSUE THREAD

**Issue by mariahadams-kr** [ant-design/ant-design#31284]
- [X] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/trigger-mode-antd4166-forked-cnh76?file=/index.js)

### Steps to reproduce
Run sandbox project linked above, use `tab` to put keyboard focus on DropDownMenu component labeled 'Click me'. Hit 'enter' on keyboard, dropdown menu overlay appears. Try using the up or down arrow keys to navigate the menu items.

### What is expected?
Should be able to navigate the dropdown menu items using the up and down arrow keys on keyboard like the submenu component per https://github.com/ant-design/ant-design/pull/30382.

### What is actually happening?
Must click to select a menu item after the dropdown menu has been triggered after hitting 'enter' on the keyboard.

| Environment | Info |
|---|---|
| antd | 4.16.6 |
| React | 16.14.0 |
| System | macOS Catalina 10.15.7 |
| Browser | Chrome Version 91.0.4472.114 (Official Build) (x86_64) |

---
I would also ask if the dropdown menu could be closed upon hitting `esc` to further improve keyboard accessibility. Thanks in advance!

**Comment by Kelvur** [comment]
I have the same problem ¿We need to active something to make this functionality to work?

**Comment by theIYD** [comment]
Do we have an update on fixing this ? 
This [PR](https://github.com/ant-design/ant-design/pull/30382) actually fixes it for Menu and not for dropdowns.

**Comment by khalibloo** [comment]
Seems to be a common issue with components that have partial overlays. The Popover, also has this issue.

**Comment by wworrall** [comment]
Just curious as to whether this is or will be included in the development roadmap? I am developing an app that needs to comply with [Web Content Accessibility Guidelines (WCAG) 2.1 AA](https://www.w3.org/TR/WCAG21/) and would like to use AntD

**Comment by YayoKB** [comment]
Any update on this? Keyboard users cannot use dropdowns properly at this point :/

**Comment by gregor-mueller** [comment]
Also related to this:

When using something like this (simplified):
```tsx
function Demo() {
  return <Dropdown
    menu={menuProps}
    trigger={['click']} // can also be omitted
  >
    <Button>Hi</Button>
  </Dropdown>
}
```

It is impossible to open the Dropdown via keyboard. When the Button is focused, neither <kbd>Enter</kbd> nor <kbd>Space</kbd> will work.


https://github.com/ant-design/ant-design/assets/28727932/9d082eba-c96c-4f86-b756-bf07a7d0dfb3


You can try yourself here: https://ant.design/components/dropdown#components-dropdown-demo-arrow-center (currently version 5.18.1 but the issue is also present at 5.12.8).

Maybe this could be a proposal for adding a `trigger=['keyboard']` as well, but I think, it should rather be the default behavior with an active `'click'` trigger.

**Comment by aaronpenny-sf** [comment]
I am seeing this issue as well and it's causing me to fail a lot of accessibility tests. Dropdown buttons with the default `trigger=['hover']` are focusable but clicking `enter` or `space` does not open the dropdown menus. This is a huge oversight in accessibility. Is there anyone looking into this?

**Comment by devuxer** [comment]
@afc163 This issue is critical for accessibility. Can the team give it some attention?
